test/bin_extract_test.go

```diff
@@ -0,0 +1,271 @@
+package test
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend/pkg/bkfs"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBinExtract_Mariner2(t *testing.T) {
```

Comment:
We should be able to move these tests directly into the distro test.

Reply:
Reason being this makes sure when we add new targets we don't forget to add these extra tests.